### PR TITLE
rspec coverage -> 100% and imported autocomplete libraries

### PR DESCRIPTION
**Pull request by mrubakh**
- Imported SimpleAutoComplete library (with implementation to come)
- Delete BlobValidator since it was overly complicated and did not suit our needs. Instead, I replaced it with a simple private method in the Item Model
- Rspec tests for the search and image uploads, brought test coverage to 100%

### CODE REVIEW

File: app/models/item.rb [hunk 1]
```diff
@@ -8,9 +8,17 @@ class Item < ApplicationRecord
...
     #size_range: 1..3.megabytes
     def self.search(query)
         Item.all.where("lower(name) LIKE ? OR category = ? OR lower(description) LIKE ?","%"+query+"%",Item.categories[query.capitalize],"%"+query+"%")
     end
+    
+    private 
+    def image_type
+      return unless image.attached?
+      if !image.content_type.in?(%('image/jpeg image/png 'image/gif))
+          errors.add(:image, "needs to be a jpeg or png!")
```
Comment (mrubakh) [inline app/models/item.rb:21]: just realized I need to include gif here as well.

**Review by amycsong98 — APPROVED** [review]
looks good! nice job refactoring the blob out

**Comment by amycsong98** [comment]
> looks good! nice job refactoring the blob out

just a bit confused why these libraries were added in with this PR but as long as the actual JS feature gets added in I think it's fine (but if that doesn't happen maybe we should take this out)

**Comment by mrubakh** [comment]
> > looks good! nice job refactoring the blob out
> 
> just a bit confused why these libraries were added in with this PR but as long as the actual JS feature gets added in I think it's fine (but if that doesn't happen maybe we should take this out)

You're so right, those should be in a different PR

**Pull request closed without merging**